### DISCUSSION TITLE
Automatically detect when client session expire

### DISCUSSION THREAD

**Discussion by goriummaximum** [emqx/emqx#6769] [Q&A] [answered]
Hi all, 

Is there any way (events, topics, callback, rules, etc.) to automatically trigger if session of client expired (expiry interval outdated)?

I have try with $SYS client disconnect topic, but it can only inform when the client disconnected, not session expired.

I am looking forward to answer.

Thank you very much.

**Comment by hjianbo** [comment, marked as answer]
There is really no $SYS type message here to notify that the session has expired.
You could try  `webhook` or `exhook`, they push `session.terminated` event. This might be what you want

**Reply by goriummaximum** [reply]
@HJianBo Thank you very much. This is the thing I want and it works!
